src/main/java/net/mcreator/ui/component/JStringListField.java

```diff
@@ -0,0 +1,203 @@
+		back.setContentAreaFilled(false);
+		back.setCursor(Cursor.getPredefinedCursor(Cursor.HAND_CURSOR));
+		back.addActionListener(e -> {
+			if (!joinEntries) {
+				if (index <= 0)
+					index = textList.size();
+				index--;
+				refreshVisibleText();
+			}
+		});
+
+		forward.setMargin(new Insets(0, 0, 0, 0));
+		forward.setBorder(BorderFactory.createEmptyBorder(0, 5, 0, 0));
+		forward.setFocusPainted(false);
+		forward.setContentAreaFilled(false);
+		forward.setCursor(Cursor.getPredefinedCursor(Cursor.HAND_CURSOR));
+		forward.addActionListener(e -> {
+			if (!joinEntries) {
+				index++;
+				if (index >= textList.size())
+					index = 0;
+				refreshVisibleText();
+			}
+		});
+
+		edit.setOpaque(false);
+		edit.setMargin(new Insets(0, 0, 0, 0));
+		edit.setBorder(BorderFactory.createEmptyBorder());
+		edit.setContentAreaFilled(false);
+		edit.setToolTipText(L10N.t("components.string_list.edit"));
+		edit.addActionListener(e -> {
+			List<String> newList = ListEditorDialog.open(parent, textList, validator, uniqueEntries);
+			if (newList != null)
+				setTextList(newList);
+		});
+
+		TechnicalButton copy = new TechnicalButton(UIRES.get("16px.copyclipboard")) {
+			@Override public String getName() {
+				return "TechnicalButton";
+			}
+		};
+		copy.setOpaque(false);
+		copy.setMargin(new Insets(0, 0, 0, 0));
+		copy.setBorder(BorderFactory.createEmptyBorder());
+		copy.setContentAreaFilled(false);
+		copy.setToolTipText(L10N.t("components.string_list.copy"));
+		copy.addActionListener(e -> Toolkit.getDefaultToolkit().getSystemClipboard()
+				.setContents(new StringSelection(String.join(System.lineSeparator(), textList)), null));
+
+		JPanel controls = new JPanel();
+		controls.setBackground((Color) UIManager.get("MCreatorLAF.DARK_ACCENT"));
+		controls.add(edit);
+		controls.add(copy);
+
+		add("West", back);
+		add("Center", scrollPane);
+		add("East", PanelUtils.westAndEastElement(forward, PanelUtils.centerInPanel(controls)));
+	}
+
+	@Override public void setEnabled(boolean b) {
+		super.setEnabled(b);
+		edit.setEnabled(b);
+	}
+
+	/**
+	 * @param joinEntries Whether string entries should be joined using commas and shown together.
+	 * @return This field instance.
+	 */
+	public JStringListField setJoinEntries(boolean joinEntries) {
+		this.joinEntries = joinEntries;
+		back.setVisible(!joinEntries);
+		forward.setVisible(!joinEntries);
+		refreshVisibleText();
+		return this;
+	}
+
+	/**
+	 * @param uniqueEntries Whether duplicate string entries in the list are forbidden.
+	 * @return This field instance.
+	 */
+	public JStringListField setUniqueEntries(boolean uniqueEntries) {
+		this.uniqueEntries = uniqueEntries;
+		return this;
+	}
+
+	/**
+	 * @return List of string entries stored in this component.
+	 */
+	public List<String> getTextList() {
+		return textList;
+	}
+
+	/**
+	 * @param textList List of string entries to be stored in this component.
+	 */
+	public void setTextList(List<String> textList) {

```

Comment:
This could be Collection instead of list, reducing need for wrapping eg. sets to lists in eg. WorkspaceDialogs.

I would also call argument newTextList to make the code a bit more clear